C error when using string concat / interpolation in const in different files and gcc or clang compiling

### Describe the bug

I'm running into a builder error when using a comptime assignment or functions with result returns in constants in different files. The same constants work when they are kept in one file.

### Expected Behavior

Same behavior when moved into different files.

### Current Behavior

```
/tmp/v_1000/my_app.15060805251445125294.tmp.c: In function ‘_vinit’:
/tmp/v_1000/my_app.15060805251445125294.tmp.c:17742:16: error: redefinition of ‘_t1’
17742 | _result_string _t1 = os__config_dir();
      |                ^~~
/tmp/v_1000/my_app.15060805251445125294.tmp.c:17733:16: note: previous definition of ‘_t1’ with type ‘_result_string’
17733 | _result_string _t1 = os__config_dir();
      |                ^~~
builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.
```

### Reproduction Steps

Builder error:
```v
// my_app/main.v
import os

const my_const = os.config_dir() or { panic(err) } + '/my_path'

fn main() {}
```

```v
// my_app/config.v
import os

const cfg_dir = os.config_dir() or { panic(err) } + '/my_app'
```
`v -cc gcc my_app`

Works in one file:
```v
import os

const (
	my_const = os.config_dir() or { panic(err) } + '/my_path'
	cfg_dir  = os.config_dir() or { panic(err) } + '/my_app'
)

fn main() {}
```

### Possible Solution

_No response_

### Additional Information/Context

Another scenario that will error with split files is using a comptime condition:

```v
// my_app/main.v
import os

const my_const = $if prod {
	os.getenv('APPDIR') + '/my_path'
} $else {
	'/my_path'
}

// Works when it's a function that has regular return and when it's not a comptime operation:
// const my_const = os.getenv('APPDIR') + '/my_path'

fn main() {}
```

```v
// my_app/config.v - same as above.
import os

const cfg_dir = os.config_dir() or { panic(err) } + '/my_app'
```

`v -cc gcc -prod my_app`

No issues with one file:
```v
import os

const (
	my_const = $if prod {
		os.getenv('APPDIR') + '/my_path'
	} $else {
		'/my_path'
	}
	cfg_dir = os.config_dir() or { panic(err) } + '/my_app'
)

fn main() {}
```

### V version

V full version: V 0.4.0 6a4bfef.815439a

### Environment details (OS name and version, etc.)

OS: linux, Pop!_OS 22.04 LTS
Processor: 16 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz

getwd: /home/turiiya/Dev/vlang/playground
vexe: /home/turiiya/Dev/vlang/v/v
vexe mtime: 2023-08-16 05:27:47

vroot: OK, value: /home/turiiya/Dev/vlang/v
VMODULES: OK, value: /home/turiiya/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.34.1
Git vroot status: weekly.2023.33
.git/config present: true

CC version: cc (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5315,7 +5315,8 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 						dep_names: g.table.dependent_names_in_expr(field_expr)
 					}
 				} else {
-					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false)
+					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false,
+						false)
 				}
 			}
 			ast.StringLiteral {
@@ -5332,9 +5333,11 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 					|| field.expr.return_type.has_flag(.result) {
 					g.inside_const_opt_or_res = true
 					unwrap_opt_res := field.expr.or_block.kind != .absent
-					g.const_decl_init_later(field.mod, name, field.expr, field.typ, unwrap_opt_res)
+					g.const_decl_init_later(field.mod, name, field.expr, field.typ, unwrap_opt_res,
+						unwrap_opt_res)
 				} else {
-					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false)
+					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false,
+						false)
 				}
 				g.inside_const_opt_or_res = false
 			}
@@ -5376,9 +5379,20 @@ fn (mut g Gen) const_decl(node ast.ConstDecl) {
 							continue
 						}
 					}
-					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false)
+					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false,
+						false)
+				} else if field.expr is ast.InfixExpr {
+					mut has_unwrap_opt_res := false
+					if field.expr.left is ast.CallExpr {
+						has_unwrap_opt_res = field.expr.left.or_block.kind != .absent
+					} else if field.expr.right is ast.CallExpr {
+						has_unwrap_opt_res = field.expr.right.or_block.kind != .absent
+					}
+					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false,
+						has_unwrap_opt_res)
 				} else {
-					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false)
+					g.const_decl_init_later(field.mod, name, field.expr, field.typ, false,
+						false)
 				}
 			}
 		}
@@ -5534,7 +5548,7 @@ fn (mut g Gen) c_const_name(name string) string {
 	return if g.pref.translated && !g.is_builtin_mod { name } else { '_const_${name}' }
 }
 
-fn (mut g Gen) const_decl_init_later(mod string, name string, expr ast.Expr, typ ast.Type, unwrap_option bool) {
+fn (mut g Gen) const_decl_init_later(mod string, name string, expr ast.Expr, typ ast.Type, unwrap_option bool, surround_cbr bool) {
 	// Initialize more complex consts in `void _vinit/2{}`
 	// (C doesn't allow init expressions that can't be resolved at compile time).
 	mut styp := g.typ(typ)
@@ -5547,12 +5561,16 @@ fn (mut g Gen) const_decl_init_later(mod string, name string, expr ast.Expr, typ
 			init.writeln('\t_const_os__args = os__init_os_args(___argc, (byte**)___argv);')
 		}
 	} else {
-		if unwrap_option {
+		if surround_cbr {
 			init.writeln('{')
+		}
+		if unwrap_option {
 			init.writeln(g.expr_string_surround('\t${cname} = *(${styp}*)', expr, '.data;'))
-			init.writeln('}')
 		} else {
 			init.writeln(g.expr_string_surround('\t${cname} = ', expr, ';'))
+		}
+		if surround_cbr {
+			init.writeln('}')
 		}
 	}
 	mut def := '${styp} ${cname}'

--- a/vlib/v/tests/modules/consts_with_or_blocks_in_different_files/config.v
+++ b/vlib/v/tests/modules/consts_with_or_blocks_in_different_files/config.v
@@ -1,0 +1,5 @@
+module consts_with_or_blocks_in_different_files
+
+import os
+
+const a_const = os.config_dir() or { panic(err) } + '/a'

--- a/vlib/v/tests/modules/consts_with_or_blocks_in_different_files/consts_should_not_conflict_test.v
+++ b/vlib/v/tests/modules/consts_with_or_blocks_in_different_files/consts_should_not_conflict_test.v
@@ -1,0 +1,10 @@
+module consts_with_or_blocks_in_different_files
+
+import os
+
+const b_const = os.config_dir() or { panic(err) } + '/b'
+
+// Just to test the two files in this folder are compiled normally
+fn test_consts_with_or_blocks_in_different_files() {
+	assert true
+}

